### PR TITLE
Minor logger and trace improvements

### PR DESCRIPTION
**Pull request by ktrzcinx**

### CODE REVIEW

File: tools/logger/logger.c [hunk 1]
```diff
@@ -247,6 +247,7 @@ int main(int argc, char *argv[])
 				usage();
 			}
 			config.dump_ldc = 1;
+			config.version_fw = 0;
```
Comment (lgirdwood) [inline tools/logger/logger.c:250]: typo in the commit message and can you state why it make no sense.

**Review by lgirdwood — CHANGES_REQUESTED** [review]

**Comment by lgirdwood** [comment]
@ktrzcinx these also seem like fixes ? if so we should cherry-pick into v1.6.

**Comment by ktrzcinx** [comment]
@lgirdwood fix from first commit can be work-around with "./sof-logger -d path -n", second commit doesn't change functionality at all, it's rather security assert for future. 
I think logger crash after `%s` is more important fix, because old version ldc files with new logger leads to crash. The split-point was month-ago at commit 03f4a1704ea03841206bf976dc95a29bc981a1af. But when we'd like to cherry pick those change, then first commit from this PR also should be picked.

File: tools/logger/logger.c (hunk 1 above)
Comment (ktrzcinx) [reply tools/logger/logger.c:250]: I didn't spot typo, but message is already extended

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**